specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json

```diff
@@ -0,0 +1,629 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "AzureStackHCI",
+    "description": "The Microsoft.AzureStackHCI Rest API spec.",
+    "version": "2023-09-01",
+    "x-ms-code-generation-settings": {
+      "name": "AzureStackHCIClient"
+    }
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/logicalNetworks/{logicalNetworkName}": {
+      "get": {
+        "operationId": "LogicalNetworks_Get",
+        "x-ms-examples": {
+          "GetLogicalNetwork": {
+            "$ref": "./examples/GetLogicalNetwork.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/LogicalNetworkNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/LogicalNetworks"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "LogicalNetworks_CreateOrUpdate",
+        "description": "The operation to create or update a logical network. Please note some properties can be set only during logical network creation.",
+        "x-ms-examples": {
+          "PutLogicalNetwork": {
+            "$ref": "./examples/PutLogicalNetwork.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
```

Comment:
>: "azure-async-operation" [](http://example.com/codeflow?start=27&length=25)

I thought this is not right, for apis using azure async operation header - I think they are supposed to leave this blank, or use 'resource' - indicating the original URI is used to fetch the final state of the updated resource?

Reply:
This was according to guideline here: https://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options that RPaaS requires us to do so. These are async operations, which azure supports both location and azure-async-operation for final states. We use azure-async-operation which is the newer option. 

#won'tfix